File: 0554-brick-wall/0554-brick-wall.java
class Solution {
    public int leastBricks(List<List<Integer>> wall) {
        Map<Integer,Integer>map=new HashMap<>();
        int mx=0;
        for(List<Integer>l:wall){
            int curr=0;
            for(int j=0;j<l.size()-1;j++){
                curr+=l.get(j);
                map.put(curr,map.getOrDefault(curr,0)+1);
                mx=Math.max(mx,map.get(curr));
            }
            
        }
        return wall.size()-mx;
    }
}

File: 0554-brick-wall/0554-brick-wall.cpp
class Solution {
public:
    int leastBricks(vector<vector<int>>& wall) {
        unordered_map<int,int>map;
        int mx=0;
        for(int i=0;i<wall.size();i++){
            int curr=0;
            for(int j=0;j<wall[i].size()-1;j++){
                curr+=wall[i][j];
                map[curr]++;
                mx=max(mx,map[curr]);
            }
        }
        return wall.size()-mx;
    }
};